src/test/java/org/broadinstitute/hellbender/utils/nio/NioTests.java

```diff
@@ -0,0 +1,118 @@
+import com.google.cloud.storage.StorageException;
+import com.google.cloud.storage.StorageOptions;
+import com.google.cloud.storage.contrib.nio.CloudStorageConfiguration;
+import com.google.cloud.storage.contrib.nio.CloudStorageFileSystem;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.*;
+import java.util.Arrays;
+
+/**
+ * Test GCS access via the NIO APIs.
+ */
+public final class NioTests extends BaseTest {
+
+    @Test
+    public void testGcsEnabled() {
+        FileSystem fs = FileSystems.getFileSystem(URI.create("gs://domain-registry-alpha"));
+    }
+
+    @Test
+    public void openPublicFile() throws IOException {
+        try (InputStream inputStream = Files.newInputStream(Paths.get(URI.create(
+                "gs://pgp-harvard-data-public/hu011C57/GS000018120-DID/GS000015172-ASM/manifest.all.sig")))) {
+            int firstByte = inputStream.read();
+        }
+    }
+
+
+    /**
+     * When we give no explicit credentials, then the system will still work if either
+     * - the GOOGLE_APPLICATION_CREDENTIALS environment variable is set
+     * - the user ran "gcloud auth login" first
+     * - the code is running on a Google Cloud Compute machine.
+     * (see http://gcloud-python.readthedocs.org/en/latest/gcloud-auth.html)
+     */
+    @Test(groups = {"cloud"})
+    public void openPrivateFileUsingDefaultCredentials() throws IOException {
+        // this file, potentially unlike the others in the set, is not marked as "Public link".
+        final String privateFile = getGCPTestInputPath() + "org/broadinstitute/hellbender/utils/private_file.txt";
+
+        Path path = Paths.get(URI.create((privateFile)));
+        int firstByte = Files.newInputStream(path).read();
+    }
+
+    /**
+     * Opening the private file even when the user is not logged in on gcloud should work
+     * when we provide explicit credentials.
+     */
+    @Test(groups = {"cloud"})
+    public void openPrivateFileWithExplicitCredentials() throws IOException {
+        // this file, potentially unlike the others in the set, is not marked as "Public link".
+        final String privateFile = getGCPTestInputPath() + "org/broadinstitute/hellbender/utils/private_file.txt";
+        final String[] split = privateFile.split("/");
+        final String BUCKET = split[2];
+        final String pathWithoutBucket = String.join("/", Arrays.copyOfRange(split, 3, split.length));
+
+        FileSystem fs = getAuthenticatedGcs(BUCKET);
+        Path path = fs.getPath(pathWithoutBucket);
+        int firstByte = Files.newInputStream(path).read();
+    }
+
+    /**
+     * Using explicit credentials only works on that access, they are not kept.
+     * This test will fail if default credentials are available.
+     */
+    @Test(enabled = false, groups = {"cloud"}, expectedExceptions = {StorageException.class})
```

Comment:
I'm keeping this test so we can run it manually every once in a while to make sure explicit auth works correctly. It's important to be able to test this, even though our Travis setup currently cannot (because it has valid default credentials). Ideally we'd set up a separate Travis run that has no default credentials, so we can avoid the predictable issue of untested features that stop working.